Doc/reference/ldap.rst

```diff
@@ -40,6 +40,14 @@ This module defines the following functions:
    when using multiple URIs you cannot determine to which URI your client
    gets connected.
 
+   If *fileno* parameter is given the socket file descriptor will be used
+   to connect to an LDAP server. The socket file descriptor must already be
+   connected. :class:`~ldap.ldapobject.LDAPObject` does not take ownership
+   of the file descriptor. It must be kept open during operations and
+   explicitly closed after the :class:`~ldap.ldapobject.LDAPObject` is
+   unbound. The internal connection type is determined from the URI, ``TCP``
+   for ``ldap://`` / ``ldaps://``, ``IPC`` for ``ldapi://``.
+
```

Comment:
Would it not be better to allow closing the descriptor automatically when the object is unbound?

Reply:
In my experience it's less flexible and causes more trouble. You can't even use ``with socket.socket()`` because socket.close() will fail with an exception when the FD has been closed outside the socket code.